vavr/src/test/java/io/vavr/AbstractValueTest.java

```diff
@@ -218,6 +218,25 @@ public void shouldReturnValueWhenGetOrNullOfNonEmpty() {
         assertThat(of(1).getOrNull()).isEqualTo(1);
     }
 
+    // -- as
+
+    @TestTemplate
+    public void shouldExecuteMapToCorrectly() {
+        assertThat(empty().mapTo(1)).isEqualTo(empty().mapTo(2));
+        assertThat(of(2).mapTo(1)).isEqualTo(of(3).mapTo(1));
+        assertThat(of(2).mapTo(1)).isEqualTo(of(3).map(ignored -> 1));
+        assertThat(of(3).mapTo(2)).isEqualTo(of(1).map(ignored -> 2));
+    }
+
+    // -- voided
```

Comment:
and this

let's update it for now, but I will eventually get rid of these comments and replace with nested junit test classes wherever it makes sense